### ISSUE TITLE
Ensure test set is independent of train set

### ISSUE THREAD

**Issue by marcosfelt** [sustainable-processes/ORDerly#114]
Any combination of reactants and products in the test set should not be in the training set. @dswigh added some tests for this in c4756c23a76e50baeb98c1535bc62cc624d6c608 but the code is still not working correctly.
